sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusProcessorClient.java

```diff
@@ -223,10 +222,12 @@ public synchronized void start() {
 
         receiveMessages();
 
-        // Start an executor to periodically check if the client's connection is active
-        if (this.scheduledExecutor == null) {
-            this.scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
-            scheduledExecutor.scheduleWithFixedDelay(() -> {
+        // Start a monitor to periodically check if the client's connection is active.
+        // NOTE: Schedulers.boundedElastic() is used here instead of Flux.interval() because the restart route involves
+        // tearing down multiple levels of clients synchronously. The boundedElastic is used instead of the parallel
+        // (parallel scheduler backing Flux.interval), so that we don't block any of the parallel threads.
+        if (monitorDisposable == null) {
```

Comment:
What was the issue with `newSingleThreadScheduledExecutor`? Also, we just need 1 thread to check the status of client connection. So, why do we need boundedElastic scheduler?

Reply:
Hi @ki1729  - @srnagar  and I had an offline chat. Srikanta explained the original reason to use "newSingleThreadScheduledExecutor" (STSE). STSE exposes the API "scheduleWithFixedDelay", which ensures the previous execution terminates before the executing the next. This way, there won't be any overlap in health checks.

We agree that STSE will create and keep the thread in the pool, even
if the thread is idle (unless allowCoreThreadTimeOut is set to true, which is false by default), so what Cx observed seems correct i.e. num_threads == num_sb_processor

The good thing is the _global_ boundedElasitcScheduler also exposes "scheduleWithFixedDelay", so we could use it instead of the current "schedulePeriodically". This way we won't end up with too many threads (due to _global_ object) and no overlap in health check.

Reply:
Hi @ki1729  - @srnagar  and I had an offline chat. Srikanta explained the original reason to use "newSingleThreadScheduledExecutor" (STSE). STSE exposes the API "scheduleWithFixedDelay", which ensures the previous execution terminates before the executing the next. This way, there won't be any overlap in health checks.

We agree that STSE will create and keep the thread in the pool, even
if the thread is idle (unless allowCoreThreadTimeOut is set to true, which is false by default), so what Cx observed seems correct i.e. num_threads == num_sb_processor

The good thing is the _global_ boundedElasitcScheduler also exposes "scheduleWithFixedDelay", so we could use it instead of the current "schedulePeriodically". This way we won't end up with too many threads (due to _global_ object) and no overlap in health check.

Reply:
Ok, I was exploring Scheduler for a different work item; I wanted to share the finding -

The _global_ "BoundedElasticScheduler::schedulePeriodically" API guarantees that the task (health check here) will not execute concurrently.

The "BoundedElasticScheduler::schedulePeriodically" delegates the task execution to JDK's "ScheduledExecutorService::scheduleAtFixedRate" and it's [java-doc](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ScheduledExecutorService.html#scheduleAtFixedRate(java.lang.Runnable,%20long,%20long,%20java.util.concurrent.TimeUnit)) says: 

> If any execution of this task takes longer than its period, then subsequent executions may start late but will not concurrently execute.

So the current implementation in this pr is concurrent safe.

Reply:
Thanks for validating this @anuchandy. So, we don't need any further changes and we can resolve this.